### PR TITLE
Fix C++ interface to DDMD

### PR DESCRIPTION
**Pull request by ibuclaw**
Still there are far too many things being forgotten when changes are made to the D API.

### CODE REVIEW

**Comment by ibuclaw** [comment]
And found a couple more in nspace.h

File: src/ddmd/attrib.d [hunk 1]
```diff
@@ -626,6 +648,35 @@ extern (C++) final class AlignDeclaration : AttribDeclaration
         return createNewScope(sc, sc.stc, sc.linkage, sc.cppmangle, sc.protection, sc.explicitProtection, this, sc.inlining);
     }
 
+    structalign_t getAlignment(Scope* sc)
```
Comment (WalterBright) [inline src/ddmd/attrib.d]: I believe these were deliberately turned into free functions in order to make it easier to use the compiler as a library. Please don't revert this.

Comment (andralex) [reply src/ddmd/attrib.d]: ping @ibuclaw cc @RazvanN7

Comment (RazvanN7) [reply src/ddmd/attrib.d]: Indeed, Walter is correct, please do not revert this.

**Comment by wilzbach** [comment]
Ping @ibuclaw 
Do you need this to be part of 2.078?

**Comment by ibuclaw** [comment]
I gave up trying to reseat gdc on 2.077 for the moment, there was just too much broken in the internals, and it seems like 2.078 has continued the trend with the recent rewrite of visitor.h that I had to do.

**Comment by ibuclaw** [comment]
Changed base from stable to master.  Added test for at least calling `semantic(Dsymbol*)` from C++.

**Comment by ibuclaw** [comment]
@wilzbach - This can be merged.

**Pull request merged**